src/common/transformations/src/transformations/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -135,6 +136,10 @@ bool shared_node_optimization(const shared_ptr<Model>& model) {
                         if (visited_nodes[j])
                             continue;
                         const auto& child_op = shared_nodes[j];
+                        if (ov::as_type_ptr<ov::op::v5::Loop>(root_op)) {
```

Comment:
please add a comment explaining why we skip this
also I believe we need to skip all ops inherited from MultiSubgraph ops

Reply:
Fixed.